### PR TITLE
Removed some of the UTC logic which shouldn't be there

### LINKED ISSUE #7307: [CAL-1146] Date overrides bug
[c1248a7b-cbe0-4f13-9bcd-a0a1e3ddb9fe](https://uploads.linear.app/e86bf957-d82f-465e-b205-135559f4b623/56e77d0a-640e-45fa-8228-045228a6f7cb/c1248a7b-cbe0-4f13-9bcd-a0a1e3ddb9fe)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1146](https://linear.app/calcom/issue/CAL-1146/date-overrides-bug)</sub>

### PR DESCRIPTION
**Pull request by emrysal**
## What does this PR do?

Fixes 11:00 - 11:00 All Day date override display issue.

* Upgrades RHF - this version supports async values so we can start getting rid of useEffect @zomars 
* Pulls in some changes from another PR
* Remove getServerSideProps
* Sorts date overrides by ascending date
* Fixed bug that caused date overrides that were still in the future to be hidden from the listing.

### CODE REVIEW

File: packages/features/schedules/components/DateOverrideList.tsx [hunk 1]
```diff
@@ -50,7 +48,7 @@ const DateOverrideList = ({
                 day: "numeric",
               }).format(item.ranges[0].start)}
             </h3>
-            {item.ranges[0].end.getUTCHours() === 0 && item.ranges[0].end.getUTCMinutes() === 0 ? (
+            {item.ranges[0].end.getHours() === 0 && item.ranges[0].end.getMinutes() === 0 ? (
```
Comment (emrysal) [inline packages/features/schedules/components/DateOverrideList.tsx:51]: getHours/getMinutes being 0-0 is only possible during an all-day event.

**Comment by CarinaWolli** [comment]
It shows 7:00 PM - 7:00 PM now for me 🤔
Screenshot 2023-02-27 at 11 13 14

File: packages/trpc/server/routers/viewer/availability.tsx [hunk 2]
```diff
@@ -157,15 +92,51 @@ export const availabilityRouter = router({
...
+            a.map((startAndEnd) => ({
+              ...startAndEnd,
+              // Turn our limited granularity into proper end of day.
+              end: new Date(startAndEnd.end.toISOString().replace("23:59:00.000Z", "23:59:59.999Z")),
+            }))
+          ),
           timeZone: schedule.timeZone || user.timeZone,
+          dateOverrides: schedule.availability.reduce((acc, override) => {
+            // only iff future date override
+            if (!override.date || override.date < new Date()) {
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/availability.tsx:113]: I can save date overrides for march 1st and 2nd (my date/time saving was march 1st at 11PM GMT-5). It is saved successfully in the database however the date overrides don't show up in the list. 

That's why:
- `new Date()` is utc so it is already march 2nd
- `override.date` has always timestamp 00:00 so it is also smaller than the utc time which was in my case 2023-03-02T04:00

**Review by zomars — APPROVED** [review]
Async values looks like a nice improvement. Thanks @emrysal 🙏🏽

File: packages/trpc/server/routers/viewer/availability.tsx (hunk 2 above)
Comment (emrysal) [reply packages/trpc/server/routers/viewer/availability.tsx]: Nice find this bug! Fixed it now ⛅

**Review by PeerRich — APPROVED** [review]

**Pull request merged**